reccmp/tools/project.py

```diff
@@ -0,0 +1,444 @@
+        SomeClass::~SomeClass() {{
+        }}
+
+        // GLOBAL: {target_id} 0x10102000
+        // STRING: {target_id} 0x10101f00
+        const char* g_globalString = "A global string";
+
+        {entry_function}
+    """
+    )
+
+
+def get_default_project_config(targets: dict[str, Path]):
+    return {
+        "targets": {
+            target_id: {"filename": path.name, "source-root": "."}
+            for target_id, path in targets.items()
+        },
+    }
+
+
+def get_default_user_config(targets: dict[str, Path]):
+    return {
+        "targets": {
+            target_id: {"path": str(path.resolve())}
+            for target_id, path in targets.items()
+        }
+    }
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Project management", allow_abbrev=False
+    )
+    parser.set_defaults(action=None)
+    parser.add_argument(
+        "--version", action="version", version=f"%(prog)s {reccmp.VERSION}"
+    )
+    parser.add_argument(
+        "-C",
+        type=Path,
+        dest="cwd",
+        metavar="<path>",
+        default=Path.cwd(),
+        help="Run as if %(prog)s was started in %(metavar)s",
+    )
+    subparsers = parser.add_subparsers()
+
+    create_parser = subparsers.add_parser("create")
+    create_parser.set_defaults(action="CREATE")
+    create_parser.add_argument(
+        "--originals",
+        type=Path,
+        nargs="+",
+        metavar="ORIGINAL",
+        required=True,
+        help="Path(s) of original executable(s)",
+    )
+    create_parser.add_argument(
+        "--path",
+        metavar="<project-directory>",
+        dest="create_directory",
+        type=Path,
+        default=Path.cwd(),
+        help="Location where to create reccmp project",
+    )
+    create_parser.add_argument(
+        "--cmake-project",
+        action="store_true",
+        dest="detect_cmake",
+        help="Create minimal CMake project",
+    )
+
+    detect_parser = subparsers.add_parser("detect")
+    detect_parser.set_defaults(action="DETECT")
+    detect_parser.add_argument(
+        "--search-path",
+        nargs="+",
+        dest="detect_search_path",
+        type=Path,
+        metavar="<path>",
+        default=[Path.cwd()],
+        help="Directory in which to look for original binaries",
+    )
+    detect_parser.add_argument(
+        "--what",
+        choices=(DetectWhat.ORIGINAL, DetectWhat.RECOMPILED),
+        type=DetectWhat,
+        default=DetectWhat.ORIGINAL,
+        dest="detect_what",
+        help="Detect original or recompiled binaries (default is original)",
+    )
+
+    argparse_add_logging_args(parser=parser)
+
+    args = parser.parse_args()
+
+    argparse_parse_logging(args=args)
+
+    if args.action == "CREATE":  # FIXME: use enum or callback function
+        return create_project(
+            project_directory=args.create_directory,
+            original_paths=args.originals,
+            cmake=args.create_cmake,
```

Comment:
Should that be `args.detect_cmake` to match the argument on 385?

Reply:
Yes, I changed it to `args.create_cmake`.